File: leetcode/js/count-tested-devices-after-test-operations.js
/**
 * @param {number[]} batteryPercentages
 * @return {number}
 */
var countTestedDevices = function (batteryPercentages) {
  let result = 0;

  for (let i = 0; i < batteryPercentages.length; i++) {
    const cur = batteryPercentages[i];

    if (cur > 0) {
      for (let j = i + 1; j < batteryPercentages.length; j++) {
        batteryPercentages[j] = Math.max(0, batteryPercentages[j] - 1);
      }

      result++;
    }
  }

  return result;
};

File: leetcode/rust/count-tested-devices-after-test-operations.rs
impl Solution {
    pub fn count_tested_devices(battery_percentages: Vec<i32>) -> i32 {
        let mut result = 0;
        let mut dec = 0;

        for p in battery_percentages {
            if p - dec > 0 {
                result += 1;
                dec += 1;
            }
        }

        result
    }
}
